### PR TITLE
Feature - New Table Component (CTE-413)

### PR DESCRIPTION
**Pull request by gustavookamoto**
**Describe pull-request**  
Contributing with a new table component. The component will be build with sort and filter included on the table header, also have options to emit events for edit and delete options.

**How to test**  
Those are the following Props:

header (required): Prop that defines which data will be shown
[{key: '', description: ''}];

content (required): Array with the data;
[]

tblClass: set the table class;
string

hasEdit: Prop to set if want to use the edit option
bool

optEdit: Prop that will be listening the edit event emitter;
EventEmitter

hasDelete: Prop to set if want to use the delete option
bool

optDelete: Prop that will be listening the delete event emitter;
EventEmitter

**Screenshots**  
js - c-table example
js - c-table example result
angular - c-table example
angular - c-table example ts

### CODE REVIEW

**Comment by helloimela** [comment]
Related issue : #551

File: src/components/table/table.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+          </div>
+        </div>
+      </td>
+  }
+  
+  /* Makes the callback for the developer */
+  private callbackDropdown(event: string, rowIndex: number)
+  {
+    debugger;
```
Comment (helloimela) [inline src/components/table/table.tsx:211]: Please remove debugger

File: src/components/table/table.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  .sort-desc:hover {
+    cursor: pointer !important;
+    background-color: white;
+    color: #041e42;
+  }
+  
+  .sort-asc:hover {
+    cursor: pointer !important;
+    background-color: white;
+    color: #041e42;
```
Comment (mmexvr) [inline src/components/table/table.scss:21]: Some css that are added, for example color: #041e42, are css that should be added in scania-theme. **All scania** related css should be added in scania-theme repo https://github.com/scania/scania-theme . Only the default css should be left.

Comment (mmexvr) [reply src/components/table/table.scss:21]: Just make a fork from scania-theme and create a branch :)

File: src/components/table/table.tsx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    if(event === "delete")
+      this.optDelete.emit(obj);
+  }
+  
+  //
+
+  render() {
+      if(!this.isArray) {
+        return <div>
+            <h1>Wrong header/content format. Please contact Customer Support</h1>
```
Comment (helloimela) [inline src/components/table/table.tsx:226]: Can you change the text to:
Wrong header/content format. Please check documentation on https://digitaldesign.scania.com/components

**Comment by helloimela** [comment]
Great work! I think it looks good overall, just small changes required before we can merge them :)

**Comment by mmexvr** [comment]
#569 PR contains some improvements and fixes for this component, please have a look at specific changes 

@gustavookamoto

**Pull request merged**